src/orchard/clojuredocs.clj

```diff
@@ -53,17 +54,20 @@
         (finally
           (.disconnect conn))))))
 
-(defn load-docs-if-not-loaded!
-  "Load exported docs from bundled or cached file when no docs are loaded.
+(let [lock (ReentrantLock.)]
```

Comment:
Isn't it a bit weird to have the lock around a definition? 

I'd also suggest adding your comment from the issue here for posterity's sake.

Reply:
+ on the comment.

Does the "function-private global variable" pattern look weird to you? I thought it was pretty common.

Reply:
> Does the "function-private global variable" pattern look weird to you?

I haven't seen it much, but I get why you did it like this. It's fine by me.